### PR TITLE
implemented smooth scroll for tips on top of sonata app

### PR DESCRIPTION
**Pull request by TahirHameed74**
implemented smooth scroll for tips on top of sonata app

### CODE REVIEW

**Comment by TahirHameed74** [comment]
Screenshot 2024-07-01 at 4 20 23 AM

File: components/TipsList.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          const formattedTips: any = await Promise.all(
+            result.tips.map(async (tip: any) => {
+              const senderUser = await getUser(tip.sender);
+              const receiverUser = await getUser(tip.receiver);
+              return {
+                text: `@${senderUser.username} tipped ${tip.amount} notes to @${receiverUser.username}`,
+                imgUrl: senderUser.pfp_url,
+              };
+            }),
+          );
```
Comment (SweetmanTech) [inline components/TipsList.tsx:25]: can you move this to a `lib` - `getFormattedTips`?

File: components/TipsList.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        }
+      } catch (error: any) {
+        setError(error || 'Unknown error');
+      }
+    };
+
+    fetchTips();
+  }, []);
+
+  if (error) return <p>Error: {error}</p>;
```
Comment (SweetmanTech) [inline components/TipsList.tsx:39]: please only include `one` return statement per react component
- do not return multiple times

File: components/TipsList.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import getUser from '@/lib/sonata/getUser';
+import React, { useEffect, useState } from 'react';
+import HorizontalScroller from './scroller/HorizontalScroller';
+
+const TipsList = () => {
+  const [tips, setTips] = useState([]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
```
Comment (SweetmanTech) [inline components/TipsList.tsx:9]: please move the useEffect to a `hook`
- suggestion: `useTipBanner`

File: components/scroller/HorizontalScroller.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        left: startOffset,
+      });
+    };
+
+    const interval = setInterval(autoScroll, 16);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
```
Comment (SweetmanTech) [inline components/scroller/HorizontalScroller.tsx:27]: please move the `ref` and `useEffect` to a hook
- suggestion: `useHorizontalScroller`

**Review by SweetmanTech — APPROVED** [review]

**Pull request merged**